src/main/java/data/JsonConverter.java

```diff
@@ -122,14 +122,44 @@ public String getConfigMap() {
     return gson.toJson(map);
   }
 
+  /**
+   * Converts a JSON array of Account objects to an ArrayList of Accounts
+   * @param accountArray JSON representation of array of account objects
+   * @return list of accounts
+   */
+  public static ArrayList<Account> buildAccountsFromJsonArray(JSONArray accountArray) {
+    ArrayList<Account> accounts = new ArrayList<>();
+    for (int i = 0; i < accountArray.length(); i++) {
+      accounts.add(new Account(accountArray.getJSONObject(i)));
+    }
+    return accounts;
+  }
+
   /** Return list of vendor IDs that exist in the filesystem.*/
   private ArrayList<String> getVendorIDs() {
     File root = new File(basePath);
     return new ArrayList<>(Arrays.asList(Objects.requireNonNull(root.list())));
   }
 
-  /** TODO Return list of account IDs corresponding to a given vendor. */
-  private ArrayList<String> getAccountIDs(String vendorID) {
-    return new ArrayList<>(Arrays.asList("ECHO", "FOXTROT", "GOLF", "HOTEL"));
+  private ArrayList<String> getAccountIDs(String vendorID) throws IOException {
+    try {
+      String config = getConfig(vendorID);
+      Vendor vendor = new Vendor(config, vendorID);
+      ArrayList<Account> accounts = vendor.getAccounts();
+      ArrayList<String> accountIds = new ArrayList<>();
+
+      if (accounts.isEmpty()) {
+        // TODO: We need a better system for error handling here
```

Comment:
Why is it an error if there is a Vendor with no Accounts associated with it yet?

Reply:
You're right, just removed that part of the code.